orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/SaveServiceAccountTask.java

```diff
@@ -0,0 +1,168 @@
+import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator;
+import com.netflix.spinnaker.fiat.shared.FiatStatus;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.front50.model.FiatServiceAccount;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import retrofit.client.Response;
+
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Save a pipeline-scoped Fiat Service Account. The roles from
+ * this service account are used for authorization decisions when
+ * the pipeline is executed from an automated trigger.
+ */
+
+@Component
+@Slf4j
+public class SaveServiceAccountTask implements RetryableTask {
+
+  private static final String SERVICE_ACCOUNT_SUFFIX = "@managed-service-account";
+
+  @Autowired(required = false)
+  private FiatStatus fiatStatus;
+
+  @Autowired(required = false)
+  private Front50Service front50Service;
+
+  @Autowired(required = false)
+  private FiatPermissionEvaluator fiatPermissionEvaluator;
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.SECONDS.toMillis(1);
+  }
+
+  @Override
+  public long getTimeout() {
+    return TimeUnit.SECONDS.toMillis(30);
+  }
+
+  @Nonnull
+  @SuppressWarnings("unchecked")
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    if (!fiatStatus.isEnabled()) {
+      throw new UnsupportedOperationException("Fiat is not enabled, cannot save permissions.");
+    }
+
+    if (front50Service == null) {
+      throw new UnsupportedOperationException(
+        "Front50 is not enabled, no way to save pipeline. Fix this by setting front50.enabled: true");
+    }
+
+    if (!stage.getContext().containsKey("pipeline")) {
+      throw new IllegalArgumentException("pipeline context must be provided");
+    }
+
+    if (!(stage.getContext().get("pipeline") instanceof String)) {
+      throw new IllegalArgumentException(
+        "'pipeline' context key must be a base64-encoded string: Ensure you're on the most recent version of gate");
+    }
+
+    Map<String, Object> pipeline;
+    try {
+      pipeline = (Map<String, Object>) stage.decodeBase64("/pipeline", Map.class);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException("pipeline must be encoded as base64", e);
+    }
+
+    // Since no permissions are specified, the pipeline is either not using triggers
```

Comment:
You're leaving me hanging after "either not using triggers" (something's missing)